### PR TITLE
pscale: Init at 0.51.0

### PR DESCRIPTION
**Pull request by pimeys**
Introducing the PlanetScale CLI, a tool to manage PlanetScale databases.

###### Motivation for this change

To be able to manage PlanetScale databases conveniently from NixOS.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cole-h** [comment]
@ofborg eval

(Sorry, ofborg's certs recently had an issue, meaning that it was unable to
react to PRs. This has been fixed, but all PRs filed in this period will need
ofborg to be manually kicked off.)

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8192,6 +8192,12 @@
     githubId = 1179566;
     name = "Nicolas B. Pierron";
   };
+  pimeys = {
+    email = "julius@nauk.io";
+    github = "pimeys";
+    id = 34967;
+    name = "Julius de Bruijn";
+  };
```
Comment (hagl) [inline maintainers/maintainer-list.nix:8200]: This should go in a separate commit titled “maintainers: add pimeys”: https://nixos.org/manual/nixpkgs/unstable/#var-meta-maintainers

Comment (pimeys) [reply maintainers/maintainer-list.nix:8200]: My apologizes! Here's the separate commit: https://github.com/NixOS/nixpkgs/pull/129005/commits/190206320cedfdf61698bcf0b47bb3d45c502b60

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -8192,6 +8192,12 @@
     githubId = 1179566;
     name = "Nicolas B. Pierron";
   };
+  pimeys = {
+    email = "julius@nauk.io";
+    github = "pimeys";
+    id = 34967;
```
Comment (lukegb) [inline maintainers/maintainer-list.nix:8198]:
```suggestion
    githubId = 34967;
```

File: pkgs/development/tools/pscale/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  vendorSha256 = "sha256-3VP2fluQLZs4nWT3O6NmCFxrqKw0/j3ASNxtbXHlZEA=";
+
+  meta = with lib; {
+    homepage = "https://www.planetscale.com/";
+    changelog = "https://github.com/planetscale/cli/releases/tag/v${version}";
+    description = "The CLI for PlanetScale Database";
+    license = licenses.asl20;
+    maintainers = with lib.maintainers; [ pimeys ];
```
Comment (hrhino) [inline pkgs/development/tools/pscale/default.nix:21]: technically you only need `with maintainers; [ pimeys ];` here, which seems to be what's generally done...

File: pkgs/development/tools/pscale/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildGoModule, fetchFromGitHub, ... }:
```
Comment (hrhino) [inline pkgs/development/tools/pscale/default.nix:1]: do you need the `...`?

File: pkgs/development/tools/pscale/default.nix (hunk 3 above)
Comment (pimeys) [reply pkgs/development/tools/pscale/default.nix]: fixed.

File: pkgs/development/tools/pscale/default.nix (hunk 4 above)
Comment (pimeys) [reply pkgs/development/tools/pscale/default.nix]: no, fixed.

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (pimeys) [reply maintainers/maintainer-list.nix]: fixed!

**Comment by hrhino** [comment]
Result of `nixpkgs-review pr 129005` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pscale</li>
  </ul>
</details>

**Comment by hrhino** [comment]
I'm not sure if I did it right, but I did
```
$ pscale signup
...
$ pscale auth login
...
$ pscale database create blurgh
Database blurgh was successfully created.
$ pscale connect blurgh
Secure connection to database blurgh and branch main is established!.

Local address to connect your application: 127.0.0.1:3306 (press ctrl-c to quit)
```
and then got
```
$ mongo --shell 127.0.0.1:3306/blurgh
MongoDB shell version v3.4.24
connecting to: mongodb://127.0.0.1:3306/blurgh
2021-07-04T17:30:13.773-0400 E QUERY    [thread1] Error: Response ID did not match the sent message ID. :
connect@src/mongo/shell/mongo.js:240:13
@(connect):1:6
exception: connect failed
```
and I'm not sure if that's because I screwed it up somehow, or if this is upstream weirdness, or if it's actually something to do with this derivation -- looks fine to me anyways, just a note

**Review by hrhino — APPROVED** [review]
